plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -294,6 +303,8 @@ public ExchangeSourceStatistics getExchangeSourceStatistics(ExchangeSourceHandle
     @Override
     public void close()
     {
-        exchangeStorage.deleteRecursively(getExchangeDirectory());
+        for (int i = 0; i < baseDirectories.size(); ++i) {
+            exchangeStorage.deleteRecursively(getExchangeDirectory(i));
```

Comment:
nit: passing `i` as `taskPartitionId` is somewhat ugly, and exploits the implementation details of `getExchangeDirectory` (that it uses `partitionId` modulo `number of base directories` to determine final directory shape).
A cleaner approach would be to have explicitly method `List<URI> getAllExchangeDirectories()` and use it in `close` and in `initialize`

Reply:
With the last commit, this part has been changed to delete all task output directories. I will create a follow up PR to delete the directories in batches.